src/coreclr/jit/assertionprop.cpp

```diff
@@ -5810,6 +5870,35 @@ ASSERT_VALRET_TP Compiler::optGetVnMappedAssertions(ValueNum vn)
     return BitVecOps::UninitVal();
 }
 
+//------------------------------------------------------------------------
+// optGetEdgeAssertions: Given a block and its predecessor, get the assertions
+//                       the predecessor creates for the block.
+//
+// Arguments:
+//      block     - The block to get the assertions for.
+//      blockPred - The predecessor of the block (creating the assertions).
+//
+// Return Value:
+//      The assertions we have about the value number.
+//
+ASSERT_VALRET_TP Compiler::optGetEdgeAssertions(const BasicBlock* block, const BasicBlock* blockPred) const
+{
+    if (blockPred->KindIs(BBJ_COND) && blockPred->FalseTargetIs(block))
+    {
+        return blockPred->bbAssertionOut;
+    }
+
+    if ((blockPred->KindIs(BBJ_ALWAYS) && blockPred->TargetIs(block)) ||
```

Comment:
I thought for `BBJ_ALWAYS` the right set was `bbAssertionOut`?

Reply:
Actually for all other BBJ_KINDs you can use `bbAssertionOut`. Just BBJ_COND is special.

Reply:
@AndyAyersMS are you sure? e.g. what about BBJ_SWITCH or some EH related blocks? I copied this function from rangecheck

Reply:
I've partially addressed your feedback as I was not confident I can just unconditionally return `bbAssertionOut` for something other than BBJ_ALWAYS/BBJ_COND(false edge). 
However, it looks like we don't typically work with anything but these two anyway.

Reply:
@AndyAyersMS I see, ok changed